src/posix/waiter.cpp

```diff
@@ -26,11 +26,20 @@ dstime Waiter::ds;
 
 PosixWaiter::PosixWaiter()
 {
-    // pipe to be able to leave the select() call
-    if (pipe(m_pipe) < 0)
+    for (;;)
     {
-        LOG_fatal << "Error creating pipe";
-        throw std::runtime_error("Error creating pipe");
+        // pipe to be able to leave the select() call
+        if (pipe(m_pipe) < 0)
+        {
+            LOG_fatal << "Error creating pipe";
+            throw std::runtime_error("Error creating pipe");
+        }
+
+        // On MacOS, pipe() keeps giving us 3 as the first fd.  Possibly on the first pipe() call per thread?
+        if (m_pipe[0] > 3 && m_pipe[1] > 3) break;
```

Comment:
With more than 2 threads in place, we could hit the case where thread1 gets the fd=3 and both thread2 and thread3 gets the fd=4 <-- issue not solved, but depending on the race condition